src/test/java/com/joyent/manta/archiver/ManataClientSupplierTest.java

```diff
@@ -0,0 +1,296 @@
+    public void notRegisteredKey() throws Exception {
+        String keyFileName = "ValidButNotRegistered.ida";
+        keyFileName = String.format("%s%s", System.getProperty("java.io.tmpdir"), keyFileName);
+        File keyFile = new File(keyFileName);
+        keyFile.createNewFile();
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(keyFileName))) {
+            String validKey = "-----BEGIN RSA PRIVATE KEY-----\n"
+                    + "MIIEpAIBAAKCAQEAqPX5IXovNwXe5ouz0TCekkdDNXS7E5dIZLmjRK4TxQAbulPj\n"
+                    + "Y0/FVclIDpjwVFw1mCdCPVNk47/++kUDAl8Gl0EUrGqeDe/Q5eK234oavz3oylWd\n"
+                    + "elmYqApH8fplMAVfeur8e+5pQHB2FLCcsgEK+3zRtfI7vgXY6xE3WUGlCmNjN/wH\n"
+                    + "jccn8G11RMibIpkFuE/g6OBm0GaC8Ikjn5lpVQcLTqRxHyHHo/0jE9rUy8IvVonb\n"
+                    + "ZaSlx7clwm6szN6aE5/FCEwaGsSxdlkzgPDc+0pO+GTdwhGs7Rl2ExcLhSWnPjUA\n"
+                    + "ZuZLcmFa4adD+NQHbTw2HlLqaHRJV4g0BsY2oQIDAQABAoIBAQCnzaLf3LmHrAz0\n"
+                    + "a0rrN55FKQFW1df2XQlJABVm4HxB6xmetDHhMBiMWpt14+7L2chglJz0yx4oE0bo\n"
+                    + "yCF0+WtSTRB7LGhM7yBJMCDvYfmudY39ZYpBOTqjqZJKgKR+TNfG/BpF+0IM/aRI\n"
+                    + "aB83qlF98zlTuoAa+TO2QiL2QnvnE7CyTn1uhhzRCT2YTuwUECqSTNHHHQrgGyCE\n"
+                    + "GD7YlzckUt1GD3cOKKUt9BE6/i2TjMSkC6N+Az1T8uzybENWupAqizcCws2jFtXS\n"
+                    + "iJmWzB3BtLHEVZN1D56wQTY2Q2FTT/iFbLDLt8Y+YW6jC1F03iHhyqaseZeUSKuS\n"
+                    + "/xwucDABAoGBANlvN74e11cZLLhmn5tbzZFUWlUcyO0x5uF84aIr9uGnZdlhj5Xt\n"
+                    + "84HzWSavbFAiRpuHwY6VudAq7cRAx+P5pCWSH3soS0kThd15p3LoEibPi3ad8gq6\n"
+                    + "vAhNEPkEqkBoZVRj4VgpweV7gdIJbIZ8JbZPU+Dq9PetseRZqqYMoyvJAoGBAMbt\n"
+                    + "xdxzM7xcQ3L7T37CBt251G3Dmcuw+fbp/dWWCKqvWjQX8yMRVtlcd28+XHyJiVOt\n"
+                    + "vjOaJgHS9/IZVwd95zv5dtfKYSk7tivtOD/YJ8FxprqtfMyxnIz2wFhWsawPuDjX\n"
+                    + "aVf4Q0oP46MBiRIQgWvMraYJtMafssnSpz/Dt3AZAoGBAJ6vGVpqPbQ2Djohwzfp\n"
+                    + "vtPiYO6ezFC3S42iyzTEqy+yMJV+KwE7oKxlQdoGyqCM80TMxcjeorY2rkG9GWTa\n"
+                    + "qx40Tz9df9w8IEUrZLZqgdzLOTf/O0bzUwkn3UwvSGUrC1CUeEAUcYqeIXd9IzPe\n"
+                    + "5NLxgAC02MTtgddqTS1UKb1pAoGANgYAw/utQywzTRie4CfFQZXj8OM78yte1wV4\n"
+                    + "3/Zc6C2y647NguqEkYchEF75MwEPAGCg1Na6F6i5mU/0aJ5ym8EF21ikxlPnB0rn\n"
+                    + "Cb+kHE7HHs9aoyRhBY9FcTgqDDZAq38kprVPYN+rzGrwVK2S2Dm/tuXP6FkabuD8\n"
+                    + "dr6qJJECgYBKc09htDe/OBPqyVtEuUv0HTA5KRNpqeheWCKTDAFtjpciYa1qLyjs\n"
+                    + "JET3gPfjWkcuDOCs6zcaj/anlr/PxdSxYw4Zu21YKZGY70ceBEJgtX5PcrgmpH7R\n"
+                    + "99+DOAhlNqwK5X+4Foo3tErRveaTczgz38G17IfoKJ7rKYHQt1o+bQ==\n" + "-----END RSA PRIVATE KEY-----";
+            bw.write(validKey);
+        } catch (Exception e) {
```

Comment:
These two lines just add noise and are not needed since the exception is being rethrown without any modification, please avoid this pattern in the future:

```java
} catch (Exception e) {
  throw e;
```